### PR TITLE
feat: add configuration widget support

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change

Add configuration widget support

Related to https://github.com/AmadeusITGroup/otter/pull/705

### CODE REVIEW

File: packages/@o3r/extractors/src/utils/config-doc.ts [hunk 1]
```diff
@@ -45,6 +48,37 @@ export function getTitleFromDocComment(docComment: DocComment): string | undefin
   return getInlineBlockTagContentFromDocComment(docComment, '@title');
 }
 
+/**
+ * Get widget information from a given DocComment.
+ *
+ *  The widget information are extracted from @widget tag.
```
Comment (cpaulve-1A) [inline packages/@o3r/extractors/src/utils/config-doc.ts:54]:
```suggestion
 * The widget information are extracted from @widget tag.
```

File: packages/@o3r/extractors/src/utils/config-doc.ts [hunk 2]
```diff
@@ -23,6 +23,9 @@ export interface ConfigDocInformation {
 
   /** Category (taken from <o3rCategories> tag) */
   categories?: CategoryDescription[];
+
+  /** Widget information (taken from @widget tag) */
```
Comment (mrednic-1A) [inline packages/@o3r/extractors/src/utils/config-doc.ts:27]: nitpicking: 'from o3rWidget tag', same below :D

File: packages/@o3r/extractors/src/utils/config-doc.ts [hunk 3]
```diff
@@ -45,6 +48,37 @@ export function getTitleFromDocComment(docComment: DocComment): string | undefin
...
+  }
+  const firstSpaceIndex = widgetInfoText.indexOf(' ');
+  if (firstSpaceIndex < 1) {
+    return;
+  }
+  const widgetType = widgetInfoText.slice(0, firstSpaceIndex);
+  let widgetParameters: ConfigPropertyWidget['parameters'] = {};
+  const widgetParametersText = widgetInfoText.slice(firstSpaceIndex + 1);
+  try {
+    widgetParameters = JSON.parse(widgetParametersText);
```
Comment (cpaulve-1A) [inline packages/@o3r/extractors/src/utils/config-doc.ts:71]: So we never validate the type of the widget parameter's value?

Comment (matthieu-crouzet) [reply packages/@o3r/extractors/src/utils/config-doc.ts:71]: Later in the extraction process there is a schema validation of what is extracted
https://github.com/AmadeusITGroup/otter/blob/main/packages/%40o3r/components/builders/component-extractor/index.ts#L87

**Review by fpaul-1A — DISMISSED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Review by cpaulve-1A — APPROVED** [review]

File: packages/@o3r/extractors/src/utils/config-doc.ts [hunk 4]
```diff
@@ -45,6 +48,46 @@ export function getTitleFromDocComment(docComment: DocComment): string | undefin
...
+export function getWidgetInformationFromDocComment(docText: string): ConfigPropertyWidget | undefined {
+  const widgetType = docText.match(/@o3rWidget (.*)/)?.[1].trim();
+  if (!widgetType) {
+    return;
+  }
+
+  const widgetParameters = Array.from(docText.matchAll(/@o3rWidgetParameter (.*)/g))
+    .map((match) => match[1].trim())
+    .reduce((acc: ConfigPropertyWidgetParameters, text) => {
+      const firstSpaceIndex = text.indexOf(' ');
```
Comment (kpanot) [inline packages/@o3r/extractors/src/utils/config-doc.ts:67]: or
```typescript
const [paramName, ...values] = test.split(' ');
if (!paramName) {
  return acc;
}
const valueText = values.join(' ');
```

**Pull request merged**